app/routes.js

```diff
@@ -1,36 +1,52 @@
 var config = require('./config');
 var ensureLoggedIn = require('connect-ensure-login').ensureLoggedIn;
 var express = require('express');
-var middleware = require('./middleware');
+var authorisation = require('./authorisation');
 var passport = require('passport');
 var proxy = require('./proxy');
 var router = new express.Router();
+const uuidv4 = require('uuid/v4');
 
 
-router.get('/login', function(req, res) {
-  res.render('login.html', {auth0: config.auth0});
-});
+const RETURN_TO = encodeURI(`${config.app.protocol}://${config.app.host}`);
+const SSO_LOGOUT_URL = `https://${config.auth0.domain}${config.auth0.sso_logout_url}?returnTo=${RETURN_TO}&client_id=${config.auth0.clientID}`;
 
-router.get('/logout', function(req, res) {
-  req.logout();
-  res.redirect('/login');
+
+router.get('/login', function(req, res, next) {
+  if (req.isAuthenticated()) {
+    if (/^http/.test(req.session.returnTo)) {
+      res.send(400, 'URL must be relative');
+    } else {
+      res.redirect(req.session.returnTo);
+    }
+  } else {
+    passport.authenticate('auth0-oidc', {
+      state: uuidv4(),
```

Comment:
Does `state` need to be passed here? If passport does not generate a `state` value by default, does it check one if supplied? Otherwise, the UUID should be stored in the session and checked in the callback.

Reply:
OK, tested without the `state` and seems to work so I'll remove this and the dependency (hopefully we'll not regret this :D)